tests/trace_server/test_clickhouse_batching.py

```diff
@@ -124,3 +130,234 @@ def test_clickhouse_batching():
             f"Expected inserts to files and call_parts tables, "
             f"but got inserts to: {insert_tables}"
         )
+
+
+def _mk_obj(project_id: str, object_id: str, wb_user_id: str, val: dict[str, Any]):
+    return tsi.ObjSchemaForInsert(
+        project_id=project_id,
+        object_id=object_id,
+        wb_user_id=wb_user_id,
+        val=val,
+    )
+
+
+def _internal_pid() -> str:
+    # Any base64 string is valid for internal project id validation
+    # Reuse the same value as other tests in this module
+    return base64.b64encode(b"test_project").decode()
+
+
+def _internal_wb_user_id() -> str:
+    return base64.b64encode(b"test_user").decode()
+
+
+def test_obj_create_batch_internal_clickhouse_insert_once():
+    """Test internal obj_create_batch performs a single ClickHouse insert."""
+    from weave.trace_server.clickhouse_trace_server_batched import ClickHouseTraceServer
+
+    mock_ch_client = MagicMock()
+    mock_ch_client.command.return_value = None
+    mock_ch_client.insert.return_value = MagicMock()
+
+    with patch.object(
+        ClickHouseTraceServer, "_mint_client", return_value=mock_ch_client
+    ):
+        server = ClickHouseTraceServer(host="test_host")
+        pid = _internal_pid()
+        wb_user_id = _internal_wb_user_id()
+        batch = [
+            _mk_obj(pid, "obj_a", wb_user_id, {"v": 1}),
+            _mk_obj(pid, "obj_b", wb_user_id, {"v": 2}),
+            _mk_obj(pid, "obj_c", wb_user_id, {"v": 3}),
+        ]
```

Comment:
why are we mocking this but later we aren't? lets just not mock if we don't need to